examples/uniswap_v2_usdc_swap.rs

```diff
@@ -0,0 +1,348 @@
+            ..
+        } => value,
+        result => panic!("'balance_of' execution failed: {result:?}"),
+    };
+
+    let balance: u128 = match <u128>::abi_decode(&value, false) {
+        Ok(balance) => balance,
+        Err(e) => panic!("'balance_of' decode failed: {:?}", e),
+    };
+
+    Ok(balance)
+}
+
+async fn get_amount_out(
+    amount_in: U256,
+    reserve_in: U256,
+    reserve_out: U256,
+    cache_db: &mut CacheDB<EmptyDBTyped<Infallible>>,
+) -> anyhow::Result<U256> {
+    let uniswap_v2_router = address!("7a250d5630b4cf539739df2c5dacb4c659f2488d");
+    sol! {
+        function getAmountOut(uint amountIn, uint reserveIn, uint reserveOut) external pure returns (uint amountOut);
+    }
+
+    let encoded = getAmountOutCall {
+        amountIn: amount_in,
+        reserveIn: reserve_in,
+        reserveOut: reserve_out,
+    }
+    .abi_encode();
+
+    let mut evm = Evm::builder()
+        .with_db(cache_db)
+        .modify_tx_env(|tx| {
+            tx.caller = address!("0000000000000000000000000000000000000000");
+            tx.transact_to = TransactTo::Call(uniswap_v2_router);
+            tx.data = encoded.into();
+            tx.value = U256::from(0);
+        })
+        .build();
+
+    let ref_tx = evm.transact().unwrap();
+    let result = ref_tx.result;
+
+    let value = match result {
+        ExecutionResult::Success {
+            output: Output::Call(value),
+            ..
+        } => value,
+        result => panic!("'get_amount_out' execution failed: {result:?}"),
+    };
+
+    let amount_out: u128 = match <u128>::abi_decode(&value, false) {
+        Ok(amount_out) => amount_out,
+        Err(e) => panic!("'get_amount_out' decode failed: {:?}", e),
+    };
+
+    Ok(U256::from(amount_out))
+}
+
+async fn get_reserves(
+    pair_address: Address,
+    cache_db: &mut CacheDB<EmptyDBTyped<Infallible>>,
+) -> anyhow::Result<(U256, U256)> {
+    sol! {
+        function getReserves() external view returns (uint112 reserve0, uint112 reserve1, uint32 blockTimestampLast);
+    }
+
+    let encoded = getReservesCall {}.abi_encode();
+
+    let mut evm = Evm::builder()
+        .with_db(cache_db)
+        .modify_tx_env(|tx| {
+            tx.caller = address!("0000000000000000000000000000000000000000");
+            tx.transact_to = TransactTo::Call(pair_address);
+            tx.data = encoded.into();
+            tx.value = U256::from(0);
+        })
+        .build();
+
+    let ref_tx = evm.transact().unwrap();
+    let result = ref_tx.result;
+
+    let value = match result {
+        ExecutionResult::Success {
+            output: Output::Call(value),
+            ..
+        } => value,
+        result => panic!("'get_reserves' execution failed: {result:?}"),
+    };
+
+    let (reserve0, reserve1): (u128, u128) = match <(u128, u128, u32)>::abi_decode(&value, false) {
+        Ok((reserve0, reserve1, _)) => (reserve0, reserve1),
+        Err(e) => panic!("'get_reserves' decode failed: {:?}", e),
+    };
```

Comment:
you should be able to just use `?` her

Reply:
I'm getting:

```
the trait `StdError` is not implemented for `alloy_sol_types::Error`, which is required by `std::result::Result<(revm::revm_primitives::alloy_primitives::Uint<256, 4>, revm::revm_primitives::alloy_primitives::Uint<256, 4>), anyhow::Error>: FromResidual<std::result::Result<Infallible, alloy_sol_types::Error>>
```

I can wrap it in `anyhow` error, but it does not look much better than current panic.

Reply:
if you want to panic, use `expect` instead

Reply:
In the end `?` works with `std` feature enabled.